interp/src/interpreter.rs

```diff
@@ -1,101 +1,114 @@
-use super::{environment::Environment, primitives};
+//! Implementation of interpreter
+//! TODO: interpreting components (e.g. in multi-component programs)
+
+use super::{environment::Environment, primitives, update::UpdateQueue};
 use calyx::{errors::FutilResult, ir};
 use std::collections::HashMap;
+use std::io;
 
 /// Evaluates a group, given an environment.
 pub fn eval_group(
     group: ir::RRC<ir::Group>,
-    env: &Environment,
+    env: Environment,
+    component: String,
 ) -> FutilResult<Environment> {
-    eval_assigns(&(group.borrow()).assignments, &env)
+    eval_assigns(&(group.borrow()).assignments, env, component)
 }
 
 // XXX(karen): I think it will need another copy of environment for each
 // iteration of assignment statements
 /// Evaluates a group's assignment statements in an environment.
+/// How this is done:
+/// First, a new write-to environment is cloned from the original read-only environment.
+/// For each clock cycle (until the group's done signal is high):
+/// Then, each assignment statement is checked for its done signal is high.
+/// If that statement's done signal is high:
+/// If the assignment is combinational, it is immediately evaluated result and stored in the write-to environment.
+/// If it is not combinational, then it is added to an update queue, to be evaluated at the end of the current clock cycle.
+/// This continues until the group's done signal is high.
 fn eval_assigns(
```

Comment:
We want this to be the structural simulator, i.e., our discrete events simulator.